source/Body.cpp

```diff
@@ -336,16 +339,18 @@ Body::initiateStep(vec6 r, vec6 rd)
 	{
 		r_ves = r;
 		rd_ves = rd;
-		return;
 	}
 	if (type == FIXED) // if the ground body, set the BCs to stationary
 	{
 		r_ves = vec6::Zero();
 		rd_ves = vec6::Zero();
-		return;
 	}
-	LOGERR << "The body is not a coupled/fixed one" << endl;
-	throw moordyn::invalid_value_error("Invalid body type");
+	else 
```

Comment:
It is a common practice to use the returns to shortcut the function execution rather than an else. It does not matter much anyway

Reply:
Do you mean return in the if blocks so as not to execute full conditional? If so, done.